src/statement_generator.cpp

```diff
@@ -517,9 +517,17 @@ unique_ptr<TableRef> StatementGenerator::GenerateSubqueryRef() {
 }
 
 unique_ptr<TableRef> StatementGenerator::GenerateTableFunctionRef() {
-	auto function = make_uniq<TableFunctionRef>();
-	auto &table_function_ref = Choose(generator_context->table_functions);
-	auto &entry = table_function_ref.get().Cast<TableFunctionCatalogEntry>();
+	auto original_val = generator_context->table_functions.size();
+	auto random_val = RandomValue(original_val);
+	auto table_function_ref = &generator_context->table_functions[random_val];
+	while (table_function_ref->get().type == CatalogType::TABLE_MACRO_ENTRY) {
+		random_val += 1 % original_val;
+		if (random_val == original_val) {
+			throw InternalException("No table_functions to test.");
+		}
+		table_function_ref = &generator_context->table_functions[RandomValue(original_val)];
```

Comment:
`RandomValue(original_val)` is still here, you can just have 

`table_function_ref = &generator_context->table_functions[random_val];` since random_val was increased by 1